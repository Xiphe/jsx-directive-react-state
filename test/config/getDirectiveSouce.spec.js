const getDirectiveSource = require('../../config/getDirectiveSource');
const { registerProvider, __reset } = require('../../config/providers');

function createNode(properties) {
  return {
    expression: {
      properties,
    },
  };
}

function createProperty(name, value) {
  return {
    key: { name },
    value: { value },
  };
}

describe('getDirectiveSource', () => {
  afterEach(() => {
    __reset();
  });

  it('throws when a undefined scope is being used', () => {
    const fakeNode = createNode([createProperty('scope', 'foo')]);

    expect(() => {
      getDirectiveSource(fakeNode, {});
    }).toThrowErrorMatchingSnapshot();
  });

  it('throws when a undefined scope type is being used', () => {
    const fakeNode = createNode([createProperty('scope', 'foo')]);

    expect(() => {
      getDirectiveSource(fakeNode, { foo: { type: 'bar' } });
    }).toThrowErrorMatchingSnapshot();
  });

  it('supports a registered scope type', () => {
    registerProvider('bar', '/bar.js');
    const fakeNode = createNode([createProperty('scope', 'foo')]);

    const source = getDirectiveSource(fakeNode, { foo: { type: 'bar' } });

    expect(source).toBe('/bar.js');
  });
});
