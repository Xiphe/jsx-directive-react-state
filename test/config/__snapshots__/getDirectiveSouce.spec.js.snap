// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`getDirectiveSource throws when a undefined scope is being used 1`] = `"State scope \\"foo\\" is not defined."`;

exports[`getDirectiveSource throws when a undefined scope type is being used 1`] = `"There is no registered provider for scope of type \\"bar\\""`;
